### PR TITLE
Moving slack webhook url to profiles.yml

### PR DESCRIPTION
**Pull request by Aylr**
## What's Here

- Moved slack webhook to profiles.yml at the top level since the notion of multiple profiles is not currently active

## Known Issues

- There's a bit in the docstring that is unknown to me.
- A real solution for environments will need to include this in a more robust way
- this is not covered by testing

### CODE REVIEW

File: docs/getting_started/pipeline_integration.rst [hunk 1]
```diff
@@ -198,7 +198,7 @@ Send Notifications
 
 The DataContext can also send notifications using a user-provided callback function based on the validation result. GE
 includes a slack-based notification in the base package. To enable a slack notification for results, simply specify
-the slack webhook in the DataContext configuration:
+the slack webhook in the ???PROFILES??? configuration:
```
Comment (Aylr) [inline docs/getting_started/pipeline_integration.rst]: I don't know what this should say here.

Comment (abegong) [reply docs/getting_started/pipeline_integration.rst]: How about "the `profiles.yml` file in `uncommitted/credentials/`"?

File: great_expectations/data_context/data_context.py [hunk 2]
```diff
@@ -1162,12 +1162,16 @@ def register_validation_results(self, run_id, validation_results, data_asset=Non
...
+        profile = self._get_all_profile_credentials()
+        if "result_callback" in profile:
+            result_callback = profile["result_callback"]
+            try:
+                slack_webhook_url = result_callback["slack"]
+                # TODO move type checking to profile/environment loaders
+                assert isinstance(slack_webhook_url, str), TypeError
+                get_slack_callback(slack_webhook_url)(validation_results)
+            except (KeyError, TypeError):
+                logger.warning("Unrecognized result_slack callback configuration. Please verify this in your profiles.yml file: {}".format(PROFILE_PATH))
```
Comment (abegong) [inline great_expectations/data_context/data_context.py:1174]: Love the informative error msg.

**Review by abegong — DISMISSED** [review]
This looks great!

It looks like we have unit test for slack, but not integration tests. Up to you whether you think it's worth it in here. The logic for firing Slack notifications (and webhooks in general) will be replaced in v0.8.0, so it might not be worth the trouble.

**Review by jcampbell — APPROVED** [review]

**Pull request merged**